src/hoc/DragDropHandler.tsx

```diff
@@ -26,49 +26,30 @@ export const DragDropHandler = ({ children, accept, index, id, moveCard, onDropI
         handlerId: monitor.getHandlerId(),
       }
     },
-    drop: (item: unknown, monitor) => {
-      if (onDropItem && item && typeof item === 'object') {
-        const hoverIndex = monitor.getClientOffset()?.y
-        if (hoverIndex !== null && ref.current && hoverIndex) {
-          onDropItem(item as IItem)
-        }
-      }
-    },
-    hover(item: unknown, monitor: DropTargetMonitor) {
-      if (!ref.current) {
-        return
-      }
-      if (!moveCard) {
-        return
-      }
-      const dragIndex = (item as IItem).index
-      const hoverIndex = index
-      const sourceId = (item as IItem).id
-      const targetId = id
 
-      if (dragIndex === hoverIndex && sourceId === targetId) {
-        return
-      }
-      const hoverBoundingRect = ref.current?.getBoundingClientRect()
-      const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2
-      const clientOffset = monitor.getClientOffset()
-      const hoverClientY = clientOffset && clientOffset.y - hoverBoundingRect.top
-      if (dragIndex < hoverIndex && hoverClientY && hoverClientY < hoverMiddleY) {
-        return
-      }
-      if (dragIndex > hoverIndex && hoverClientY && hoverClientY > hoverMiddleY) {
-        return
+    // hover(item: unknown, monitor: DropTargetMonitor) {
+    //   if (!ref.current) {
+    //     return
+    //   }
+    //   if (!moveCard) {
+    //     return
+    //   }
+    // },
+
+    drop: (item: unknown, monitor) => {
+      if (onDropItem) {
+        onDropItem({
+          taskId: (item as { taskId: string }).taskId,
+          targetWorkflowStateId: id,
+        })
       }
-      moveCard(dragIndex, hoverIndex, sourceId, targetId)
-      ;(item as IItem).index = hoverIndex
-      ;(item as IItem).id = targetId
     },
   })
 
   const [{ isDragging }, drag] = useDrag({
     type: accept,
     item: () => {
-      return { id, index }
+      return { taskId: id }
```

Comment:
It would be better to make this a custom type. I see this is used at multiple places so giving it a meaningful type name would also make it easier to understand

Reply:
I couldn't understand properly where should the type be given. Can you please explain? @rrojan